packages/block-library/src/post-featured-image/dimension-controls.js

```diff
@@ -33,14 +33,23 @@ const SCALE_OPTIONS = (
 		/>
 		<SegmentedControlOption
 			value="fill"
-			label={ _x(
-				'Stretch',
-				'Scale option for Image dimension control'
-			) }
+			label={ _x( 'Fill', 'Scale option for Image dimension control' ) }
 		/>
 	</>
 );
 
+const scaleHelp = {
+	cover: __(
+		'Image is sized to maintain its aspect ratio and cropped to fill the entire space.'
+	),
+	contain: __(
+		'Image is scaled to fill the space without clipping the content.'
```

Comment:
I wondered whether we should include that this setting maintains the aspect ratio. But the visual effect and the added context of the fill option feels sufficient.